src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -189,57 +303,139 @@ public String addEmailAlias(String alias) throws Exception
+     *
+     * @param client
+     * @param fileModel
      * @return
-     * @throws DataPreparationException
      */
-    public FileModel createContent(DocumentType documentType) throws DataPreparationException
+    public FileModel createContentV1Api(AlfrescoHttpClient client, FileModel fileModel)
     {
-        String newContent = String.format("%s.%s", RandomData.getRandomName("file"), Utility.cmisDocTypeToExtentions(documentType));
-        String newLocation = Utility.buildPath(getLastResource(), newContent);
-        STEP(String.format("DATAPREP: Creating a new non-empty content %s in %s ", newContent, getLastResource()));
+        // Build request
+        String nodeId = this.getNodeRef();
+        String reqUrl = client.getApiVersionUrl() + "nodes/" + nodeId + "/children";
+        HttpPost post  = new HttpPost(reqUrl);
+        JSONObject body = new JSONObject();
+        body.put("name", fileModel.getName());
+        body.put("nodeType", "cm:content");
 
-        if (getLastResource().isEmpty())
-            setLastResource(RandomData.getRandomName("Folder"));
+        // Set Title or Description if specified
+        if (fileModel.getTitle() != null || fileModel.getDescription() != null)
+        {
+            body.put("cm:title", fileModel.getTitle());
+            body.put("cm:description", fileModel.getDescription());
+        }
+
+        post.setEntity(client.setMessageBody(body));
+
+        // Send Request
+        logger.info(String.format("Create content with name '%s' by: ", fileModel.getName()));
+        logger.info(String.format("POST: '%s'", reqUrl));
+        HttpResponse response = client.execute(currentUser.getUsername(), currentUser.getPassword(), post);
+        if(HttpStatus.SC_CREATED == response.getStatusLine().getStatusCode())
+        {
+            JSONObject entryResponse = client.readStream(response.getEntity());
+            JSONObject entryValueMap = (JSONObject) entryResponse.get("entry");
+
+            fileModel.setNodeRef(entryValueMap.get("id").toString());
+            fileModel.setName(entryValueMap.get("name").toString());
+
+            String fileLocation = Utility.buildPath(getLastResource(), fileModel.getName());
+            fileModel.setCmisLocation(fileLocation);
+            fileModel.setProtocolLocation(fileLocation);
+
+            logger.info(String.format("Successful created content with id '%s' ", entryValueMap.get("id").toString()));
+            return fileModel;
+        }
+        else
+        {
+            logger.error(client.getParameterFromJSON(response,"briefSummary", "error"));
+            return new FileModel();
+        }
+    }
+
+    /**
+     * This method updates the content of the created document
+     *
+     * @param client
+     * @param fileModel
+     */
+    public void updateContent(AlfrescoHttpClient client, FileModel fileModel)
+    {
+        // Build request
+        String nodeId = fileModel.getNodeRef();

```

Comment:
I don't think this method handles the case where the previous call exits through the else case: 
"
 else
        {
            logger.error(client.getParameterFromJSON(response,"briefSummary", "error"));
            return new FileModel();
        }
"
It assumes the fileModel is a valid one. This may be intended, I don't know...